integration-tests/gradle/build.gradle.kts

```diff
@@ -111,6 +137,54 @@ subprojects {
         }
       }
     }
+
+    if (name.contains("Java6Constraint")) {
+      dependencies {
```

Comment:
To clarify:

1. If you do not mention Guava at all in your build script, and just get it transitively, there is nothing you need to do as long as you are fine with the default varian selection.
2. If you declare a dependency to Guava directly, there is  nothing you need to do as long as you are fine with the default varian selection.
3. If you want to select a different variant than the default, you can dot that, differently to what is shown in the test, directly in the dependency. E.g.:
```
dependencies {
  api("com.google.guava:guava:30.0+") {
    attributes { attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, 6) }
  }
}
```
This might just work **or** if there is another transitive dependency, which does not explicitly request jvm=6, you can get a conflict because Gradle selects different variants for the different dependencies. Then you need the `resolutionStrategy.capabilitiesResolution` rule to solve the conflict. In a normal build script that looks less complicated. It's only that I needed to use reflection in this test so that it can also compile with Gradle 5.
4. You can also create the situation above just in your local build script by having multiple dependencies or a dependency + a dependency constraint, on Guava in your build script where not all of them request the same attributes. (That's the case in this test.)

With the Gradle Metadata, we'll make sure that (1) and (2) are done correctly. At the moment the default is often wrong when conflicts occur (e.g. right now `29-jr` and `30-android` will conflict resolve to `30-android` for a Java project). When the default works correct, the hope is that an override is not needed very often. While right now many users have to do it, because the default behaviour resolves to the wrong version/variant.

Reply:
Now I'm thinking about what to put in the release notes. It sounds like we can mostly just say that things will get better automatically :) But in the interest of helping users who do encounter any problems:

If users have a transitive dependency on `google-collections` or `listenablefuture` in addition to some dependency on Guava proper, then they'll need to something along the lines of [these instructions](https://docs.gradle.org/current/userguide/dependency_capability_conflict.html#sub:selecting-between-candidates), which is similar to what you demonstrate above but without the need for reflection.

Do I have that right? I _think_ I'm repeating what you said back to you, but obviously it's been forever. Sorry again.

(That behavior is an improvement over having `google-collections` _also_ on the classpath. It's a more mixed bag for `listenablefuture`: Instead of having the conflict with that automatically resolved in a hacky way (which IIRC upset Android Studio builds until they put in a hack of their own for us), users have to do it explicitly.)